pkg/utils/tekton/controller.go

```diff
@@ -96,31 +139,66 @@ func (s *SuiteController) ListTaskRuns(ns string, labelKey string, labelValue st
 	return s.PipelineClient().TektonV1beta1().TaskRuns(ns).List(context.TODO(), listOptions)
 }
 
-func (k KubeController) WatchTaskPod(tr string, taskTimeout int) error {
-	trUpdated, err := k.Tektonctrl.GetTaskRun(tr, k.Namespace)
-	if err != nil {
-		return err
-	}
-	pod, err := k.Commonctrl.GetPod(k.Namespace, trUpdated.Status.PodName)
-	if err != nil {
-		return err
-	}
-	return k.Commonctrl.WaitForPod(k.Commonctrl.IsPodSuccessful(pod.Name, k.Namespace), time.Duration(taskTimeout)*time.Second)
+func (k KubeController) WatchPipelineRun(pipelineRunName string, taskTimeout int) error {
+	return k.Commonctrl.WaitForPod(k.Tektonctrl.CheckPipelineRunFinished(pipelineRunName, k.Namespace), time.Duration(taskTimeout)*time.Second)
 }
 
-func (k KubeController) GetTaskRunResult(tr *v1beta1.TaskRun, result string) (string, error) {
-	for _, trResult := range tr.Status.TaskRunResults {
-		if trResult.Name == result {
-			return trResult.Value, nil
+func (k KubeController) GetTaskRunResult(pr *v1beta1.PipelineRun, result string) (string, error) {
```

Comment:
I think this method should take an additional parameter now to specify the which task to get the results from. Otherwise, our assertions will not be as accurate.